jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/util/WebSocketDemander.java

```diff
@@ -75,18 +83,54 @@ public DemandingFlusher(IncomingFrames emitFrame)
      */
     protected abstract boolean handle(Frame frame, Callback callback, boolean first);
 
-    @Override
-    public void demand()
+    /**
+     * <p>This is used within an implementation of {@link #handle(Frame, Callback, boolean)}
+     * to forward a frame onto the next layer of processing.</p>
+     *
+     * <p>This method should only be called ONCE within each invocation of {@link #handle(Frame, Callback, boolean)}
+     * otherwise</p>
+     * @param frame the WebSocket frame.
+     * @param callback to release frame payload.
+     */
+    public void emitFrame(Frame frame, Callback callback)
     {
-        if (!_demand.compareAndSet(false, true))
-            throw new ReadPendingException();
-        iterate();
+        Throwable failure = null;
+        try (AutoLock l = _lock.lock())
```

Comment:
```suggestion
        try (AutoLock ignored = _lock.lock())
```

Using `ignored` saves warnings in the IDE. Please update also elsewhere.